File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number[]}
 */
var sumEvenAfterQueries = function(n, q) {
    let arr = [];
    let sum=0;
    for(let i=0;i<n.length;i++){
        if(!(n[i]&1)){
            sum += n[i];
        }
    }
    
    q.forEach((element)=>{
        let val = element[0];
        let x = element[1];
        let prev = n[x];
        n[x] += val;
        if(!(n[x]&1)){
            sum += val;
            if(prev&1) sum+=prev;
        }else if(!(prev&1)){
            sum-=prev;
        }
        arr.push(sum);
    })
    return arr;
};

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& n, vector<vector<int>>& q){
        vector<int> ans;
        int sum = 0;
        for(int i:n){
            if(!(i&1)){
                sum += i;
            }
        }
        for(auto i:q){
            int val = i[0],x = i[1];
            int prev = n[x];
            n[x] += val;
            if(!(n[x]&1)){
                sum += val;
                if(prev&1) sum += prev;
            }else if(!(prev&1)){
                sum -= prev;
            }
            ans.push_back(sum);
        }
        return ans;
    }
};